File: 125-valid-palindrome/valid-palindrome.js
var isPalindrome = function (s) {
  const arr = s
    .toLowerCase()
    .replace(/[\p{P}\s\$1,(`)]/gu, "")
    .split("");
  let inc = 0;
  let dec = arr.length - 1;
  while (inc < dec) {
    if (arr[inc] !== arr[dec]) {
      return false;
    }
    inc++;
    dec--;
  }
  return true;
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
            char[] arr = s.toLowerCase().replaceAll("[^a-z0-9]", "").toCharArray();
    int left = 0;
    int right = arr.length - 1;
    while (left < right) {
      if (arr[left] != arr[right]) {
        return false;
      }
      left++;
      right--;
    }
    return true;
    }
}